### PR TITLE
refactor: split `docker` job into `candidates` and `releases`

### PR DESCRIPTION
**Pull request by innovaker**

### CODE REVIEW

File: .github/workflows/containers.yml [hunk 1]
```diff
@@ -13,75 +13,111 @@ on:
...
     
 jobs:
-  docker:
+  namespaces:
+    runs-on: ubuntu-latest
+    outputs:
+      docker-hub: ${{ secrets.DOCKER_HUB_NAMESPACE || github.repository_owner }}
+      ghcr: ${{ github.repository_owner }}
+    steps:
+      - run: exit
```
Comment (petejohanson) [inline .github/workflows/containers.yml:22]: I kind of prefer `true` for cases like this, but this is fine too.

File: .github/workflows/containers.yml [hunk 2]
```diff
@@ -13,75 +13,111 @@ on:
...
+    runs-on: ubuntu-latest
+    outputs:
+      json: ${{ steps.import.outputs.json }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Import from architectures.yml
+        id: import
+        run: |
+          import yaml, json
```
Comment (petejohanson) [inline .github/workflows/containers.yml:47]: Sneaky! Handy `yaml2json` using just the system items. I dig.

File: .github/workflows/containers.yml [hunk 3]
```diff
@@ -13,75 +13,111 @@ on:
...
+      - tags
+    if: ${{ !startsWith(github.ref, 'refs/tags') }}
     runs-on: ubuntu-latest
     strategy:
       max-parallel: 1 # takes advantage of caching between jobs
-      matrix:
-        architecture:
-          - arm
+      matrix: 
+        architecture: ${{ fromJSON(needs.architectures.outputs.json) }}
```
Comment (petejohanson) [inline .github/workflows/containers.yml:92]: I dig the use of `needs` for these.

**Review by petejohanson — APPROVED** [review]
All of this looks fabulous! Easy to see how this sets things up w/ the `architectures.yml` file to build for multiple targets.

File: .github/workflows/containers.yml (hunk 1 above)
Comment (innovaker) [reply .github/workflows/containers.yml:22]: `true`?

Comment (petejohanson) [reply .github/workflows/containers.yml:22]: `true` is a standard posix shell command that just returns with a 0 exit code for success.

Comment (innovaker) [reply .github/workflows/containers.yml:22]: ahhh, I thought you were talking about the conditional logic above it!
Yeah, good idea!  I'll change it accordingly.

**Pull request merged**